src/plugins/profiling/server/routes/search_flamechart.ts

```diff
@@ -189,19 +192,52 @@ async function queryFlameGraph(
   const executableDocIDs = new Set<string>(); // Set of unique executable FileIDs.
   const stackTraceIDs = [...stackTraceEvents.keys()];
   const chunkSize = Math.floor(stackTraceEvents.size / nQueries);
-  const chunks = chunk(stackTraceIDs, chunkSize);
+  let chunks = chunk(stackTraceIDs, chunkSize);
+
+  if (chunks.length !== nQueries) {
+    // The last array element contains the remainder, just drop it as irrelevant.
+    chunks = chunks.slice(0, nQueries);
+  }
```

Comment:
AFAIU the smaller `nQueries`, the more we risk to lose a significant portion of the stacktraces queried at next step, because the remainder increases.
Why not keeping all of them?

Reply:
The remainde is the rest of the devision. So if we devide by 8, it is 1-7 ids that we lose. In comparison the other chunks have ~20000/8 = 2.500. So the 1-7 are statistically not relevant. Btw, we already dropped them before - you just didn't recognize it 😄